pandas/_libs/tslibs/offsets.pyx

```diff
@@ -3350,7 +3350,13 @@ cdef class _CustomBusinessMonth(BusinessMixin):
         """
         Define default roll function to be called in apply method.
         """
-        cbday = CustomBusinessDay(n=self.n, normalize=False, **self.kwds)
+        if self.offset:
+            cbday_kwds = self.kwds.copy()
+            cbday_kwds['offset'] = timedelta(0)
+        else:
+            cbday_kwds = self.kwds
+
+        cbday = CustomBusinessDay(n=1, normalize=False, **cbday_kwds)
```

Comment:
Changing from n=self.n to n=1 does not have any effect on the functionality. I just personally thought having n=1 is 1) more consistent with MonthBegin(n=1) and MonthEnd(n=1) below; and 2) slightly more indicative of the role of CustomBusinessDay, which is simply to apply a 1-day rollforward/rollback. Happy to change it back.